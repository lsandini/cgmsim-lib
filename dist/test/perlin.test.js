"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const perlin_1 = require("../src/perlin");
const oldPerlin_1 = require("../old/oldPerlin");
const defaultParams = {
    amplitude: 0.3,
    octaveCount: 3,
    persistence: 0.3,
    maxAbsValue: 0.05,
    seed: 'cgmsim',
    mode: 'daily'
};
describe('test perlin with fake timers', () => {
    beforeAll(() => {
        jest.useFakeTimers('modern');
    });
    beforeEach(() => {
        jest.setSystemTime(new Date('2001-01-01'));
    });
    afterAll(() => {
        jest.useRealTimers();
    });
    test('run twice in same day should return same array', () => {
        jest.setSystemTime(new Date('2022-05-30T00:01'));
        const perl = perlin_1.default(defaultParams);
        jest.setSystemTime(new Date('2022-05-30T01:01'));
        const perl2 = perlin_1.default(defaultParams);
        expect(perl[0].time).toBe(perl2[0].time);
        expect(perl[0].noise).toBe(perl2[0].noise);
    });
    test('run twice different day should return different array', () => {
        const perl = perlin_1.default(defaultParams);
        jest.setSystemTime(new Date('2001-01-02T01:01'));
        const perl2 = perlin_1.default(defaultParams);
        expect(perl[0].time).not.toBe(perl2[0].time);
        expect(perl[0].noise).not.toBe(perl2[0].noise);
    });
    test('run in same day should return same array', () => {
        const perl = perlin_1.default(defaultParams);
        expect(perl.length).toBe(288);
        expect(perl[0]).not.toBe(perl[1]);
        expect(perl).toMatchSnapshot();
    });
});
describe('test perlin with real timers', () => {
    test('run twice in same day should return same array', () => {
        const perl = perlin_1.default(defaultParams);
        setTimeout(() => {
            const perl2 = perlin_1.default(defaultParams);
            expect(perl[0].time).toBe(perl2[0].time);
            expect(perl[0].noise).toBe(perl2[0].noise);
        }, 100);
    });
});
describe('test comparing old perlin', () => {
    test('run in same day should return same array', () => {
        const perl = perlin_1.default(defaultParams);
        const oldPerl = oldPerlin_1.default();
        const countPositiveValPerl = perl.filter(p => p.noise > 0).length;
        const countPositiveValOldPerl = oldPerl.filter(p => p.noise > 0).length;
        const maxValPerl = perl.sort((a, b) => b.noise - a.noise)[0];
        const maxValOldPerl = oldPerl.sort((a, b) => b.noise - a.noise)[0];
        const minValPerl = perl.sort((a, b) => a.noise - b.noise)[0];
        const minValOldPerl = oldPerl.sort((a, b) => a.noise - b.noise)[0];
        const sumValPerl = perl.reduce((acc, el) => acc + el.noise, 0);
        const sumValOldPerl = oldPerl.reduce((acc, el) => acc + el.noise, 0);
        console.log("new", { countPositiveValPerl, maxValPerl, minValPerl, sumValPerl });
        console.log("old", { countPositiveValOldPerl, maxValOldPerl, minValOldPerl, sumValOldPerl });
        expect(perl.length).toBe(oldPerl.length);
    });
});
//# sourceMappingURL=perlin.test.js.map