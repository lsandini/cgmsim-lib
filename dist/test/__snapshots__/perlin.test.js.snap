// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test perlin with fake timers run in same day should return same array 1`] = `
Array [
  Object {
    "noise": -0.009970024799999998,
    "order": 0,
    "time": 978307200000,
  },
  Object {
    "noise": -0.01407240034676259,
    "order": 1,
    "time": 978307500000,
  },
  Object {
    "noise": -0.01982042178201439,
    "order": 2,
    "time": 978307800000,
  },
  Object {
    "noise": -0.025040177881294964,
    "order": 3,
    "time": 978308100000,
  },
  Object {
    "noise": -0.0329136083,
    "order": 4,
    "time": 978308400000,
  },
  Object {
    "noise": -0.02571176739784173,
    "order": 5,
    "time": 978308700000,
  },
  Object {
    "noise": -0.020369980381294964,
    "order": 6,
    "time": 978309000000,
  },
  Object {
    "noise": -0.02944753810935252,
    "order": 7,
    "time": 978309300000,
  },
  Object {
    "noise": -0.043726112500000004,
    "order": 8,
    "time": 978309600000,
  },
  Object {
    "noise": -0.027784724926618706,
    "order": 9,
    "time": 978309900000,
  },
  Object {
    "noise": -0.009518823297122309,
    "order": 10,
    "time": 978310200000,
  },
  Object {
    "noise": 0.0026807498474820182,
    "order": 11,
    "time": 978310500000,
  },
  Object {
    "noise": 0.013280062200000012,
    "order": 12,
    "time": 978310800000,
  },
  Object {
    "noise": 0.004312237974100713,
    "order": 13,
    "time": 978311100000,
  },
  Object {
    "noise": -0.00587549085035971,
    "order": 14,
    "time": 978311400000,
  },
  Object {
    "noise": -0.00579311466906475,
    "order": 15,
    "time": 978311700000,
  },
  Object {
    "noise": -0.004186325500000004,
    "order": 16,
    "time": 978312000000,
  },
  Object {
    "noise": -0.002190709475539572,
    "order": 17,
    "time": 978312300000,
  },
  Object {
    "noise": -0.0032720493021582767,
    "order": 18,
    "time": 978312600000,
  },
  Object {
    "noise": -0.0020429398237410065,
    "order": 19,
    "time": 978312900000,
  },
  Object {
    "noise": -0.004879230200000001,
    "order": 20,
    "time": 978313200000,
  },
  Object {
    "noise": -0.012840761082733812,
    "order": 21,
    "time": 978313500000,
  },
  Object {
    "noise": -0.028442157071942448,
    "order": 22,
    "time": 978313800000,
  },
  Object {
    "noise": -0.03002749989856115,
    "order": 23,
    "time": 978314100000,
  },
  Object {
    "noise": -0.035781201000000006,
    "order": 24,
    "time": 978314400000,
  },
  Object {
    "noise": -0.03119333884892087,
    "order": 25,
    "time": 978314700000,
  },
  Object {
    "noise": -0.024065659953956833,
    "order": 26,
    "time": 978315000000,
  },
  Object {
    "noise": -0.024032411210071943,
    "order": 27,
    "time": 978315300000,
  },
  Object {
    "noise": -0.021119189000000003,
    "order": 28,
    "time": 978315600000,
  },
  Object {
    "noise": -0.009831203600000005,
    "order": 29,
    "time": 978315900000,
  },
  Object {
    "noise": 0.007088280201438854,
    "order": 30,
    "time": 978316200000,
  },
  Object {
    "noise": 0.020664095661870496,
    "order": 31,
    "time": 978316500000,
  },
  Object {
    "noise": 0.0312466354,
    "order": 32,
    "time": 978316800000,
  },
  Object {
    "noise": 0.024878634532374114,
    "order": 33,
    "time": 978317100000,
  },
  Object {
    "noise": 0.02629065225539569,
    "order": 34,
    "time": 978317400000,
  },
  Object {
    "noise": 0.02939805327697842,
    "order": 35,
    "time": 978317700000,
  },
  Object {
    "noise": 0.03780665899999999,
    "order": 36,
    "time": 978318000000,
  },
  Object {
    "noise": 0.03364062315107913,
    "order": 37,
    "time": 978318300000,
  },
  Object {
    "noise": 0.03967939212158274,
    "order": 38,
    "time": 978318600000,
  },
  Object {
    "noise": 0.03358669312589928,
    "order": 39,
    "time": 978318900000,
  },
  Object {
    "noise": 0.033503104800000016,
    "order": 40,
    "time": 978319200000,
  },
  Object {
    "noise": 0.02486017168848921,
    "order": 41,
    "time": 978319500000,
  },
  Object {
    "noise": 0.013480991834532385,
    "order": 42,
    "time": 978319800000,
  },
  Object {
    "noise": 0.0025634761294964087,
    "order": 43,
    "time": 978320100000,
  },
  Object {
    "noise": -0.010086909999999998,
    "order": 44,
    "time": 978320400000,
  },
  Object {
    "noise": -0.0026263257323741046,
    "order": 45,
    "time": 978320700000,
  },
  Object {
    "noise": 0.0013876252431654634,
    "order": 46,
    "time": 978321000000,
  },
  Object {
    "noise": 0.016318531976259,
    "order": 47,
    "time": 978321300000,
  },
  Object {
    "noise": 0.034079108999999996,
    "order": 48,
    "time": 978321600000,
  },
  Object {
    "noise": 0.014094456625899268,
    "order": 49,
    "time": 978321900000,
  },
  Object {
    "noise": -0.004561348299280579,
    "order": 50,
    "time": 978322200000,
  },
  Object {
    "noise": -0.016681296142446046,
    "order": 51,
    "time": 978322500000,
  },
  Object {
    "noise": -0.0356562384,
    "order": 52,
    "time": 978322800000,
  },
  Object {
    "noise": -0.030218847974100724,
    "order": 53,
    "time": 978323100000,
  },
  Object {
    "noise": -0.025244099064748206,
    "order": 54,
    "time": 978323400000,
  },
  Object {
    "noise": -0.02214143512230216,
    "order": 55,
    "time": 978323700000,
  },
  Object {
    "noise": -0.026932801000000003,
    "order": 56,
    "time": 978324000000,
  },
  Object {
    "noise": -0.009113215061151078,
    "order": 57,
    "time": 978324300000,
  },
  Object {
    "noise": -0.0009047887618705008,
    "order": 58,
    "time": 978324600000,
  },
  Object {
    "noise": 0.021810904501438827,
    "order": 59,
    "time": 978324900000,
  },
  Object {
    "noise": 0.047532395199999994,
    "order": 60,
    "time": 978325200000,
  },
  Object {
    "noise": 0.03758858363309353,
    "order": 61,
    "time": 978325500000,
  },
  Object {
    "noise": 0.03030436382086331,
    "order": 62,
    "time": 978325800000,
  },
  Object {
    "noise": 0.022655428038848915,
    "order": 63,
    "time": 978326100000,
  },
  Object {
    "noise": 0.016446579000000003,
    "order": 64,
    "time": 978326400000,
  },
  Object {
    "noise": 0.007240048025179857,
    "order": 65,
    "time": 978326700000,
  },
  Object {
    "noise": -0.004377719684892083,
    "order": 66,
    "time": 978327000000,
  },
  Object {
    "noise": -0.025011813649640287,
    "order": 67,
    "time": 978327300000,
  },
  Object {
    "noise": -0.0498517413,
    "order": 68,
    "time": 978327600000,
  },
  Object {
    "noise": -0.02886149005323741,
    "order": 69,
    "time": 978327900000,
  },
  Object {
    "noise": -0.00946661881438849,
    "order": 70,
    "time": 978328200000,
  },
  Object {
    "noise": -0.01542080514172662,
    "order": 71,
    "time": 978328500000,
  },
  Object {
    "noise": -0.014610214900000001,
    "order": 72,
    "time": 978328800000,
  },
  Object {
    "noise": -0.016705602147482013,
    "order": 73,
    "time": 978329100000,
  },
  Object {
    "noise": -0.02474633131582734,
    "order": 74,
    "time": 978329400000,
  },
  Object {
    "noise": -0.023145378987050366,
    "order": 75,
    "time": 978329700000,
  },
  Object {
    "noise": -0.026756661400000003,
    "order": 76,
    "time": 978330000000,
  },
  Object {
    "noise": -0.017949501611510793,
    "order": 77,
    "time": 978330300000,
  },
  Object {
    "noise": -0.006088092829496405,
    "order": 78,
    "time": 978330600000,
  },
  Object {
    "noise": 0.0005214841503597134,
    "order": 79,
    "time": 978330900000,
  },
  Object {
    "noise": 0.011388491,
    "order": 80,
    "time": 978331200000,
  },
  Object {
    "noise": 0.010467386621582739,
    "order": 81,
    "time": 978331500000,
  },
  Object {
    "noise": 0.008098573736690637,
    "order": 82,
    "time": 978331800000,
  },
  Object {
    "noise": 0.0024093910935251706,
    "order": 83,
    "time": 978332100000,
  },
  Object {
    "noise": -0.00035916789999999865,
    "order": 84,
    "time": 978332400000,
  },
  Object {
    "noise": -0.010471897248201446,
    "order": 85,
    "time": 978332700000,
  },
  Object {
    "noise": -0.027508699319424464,
    "order": 86,
    "time": 978333000000,
  },
  Object {
    "noise": -0.03725638156258993,
    "order": 87,
    "time": 978333300000,
  },
  Object {
    "noise": -0.047003713,
    "order": 88,
    "time": 978333600000,
  },
  Object {
    "noise": -0.03201582913453238,
    "order": 89,
    "time": 978333900000,
  },
  Object {
    "noise": -0.025757348329496402,
    "order": 90,
    "time": 978334200000,
  },
  Object {
    "noise": -0.01800535164028777,
    "order": 91,
    "time": 978334500000,
  },
  Object {
    "noise": -0.007765657000000002,
    "order": 92,
    "time": 978334800000,
  },
  Object {
    "noise": -0.01611001066546762,
    "order": 93,
    "time": 978335100000,
  },
  Object {
    "noise": -0.021421821388489214,
    "order": 94,
    "time": 978335400000,
  },
  Object {
    "noise": -0.01921339190143885,
    "order": 95,
    "time": 978335700000,
  },
  Object {
    "noise": -0.026873149700000005,
    "order": 96,
    "time": 978336000000,
  },
  Object {
    "noise": -0.020657446774820152,
    "order": 97,
    "time": 978336300000,
  },
  Object {
    "noise": -0.018673953200719434,
    "order": 98,
    "time": 978336600000,
  },
  Object {
    "noise": -0.00938783081366907,
    "order": 99,
    "time": 978336900000,
  },
  Object {
    "noise": 0.001993508899999996,
    "order": 100,
    "time": 978337200000,
  },
  Object {
    "noise": 0.00490442630575539,
    "order": 101,
    "time": 978337500000,
  },
  Object {
    "noise": 0.014426321630935254,
    "order": 102,
    "time": 978337800000,
  },
  Object {
    "noise": 0.020634547497122305,
    "order": 103,
    "time": 978338100000,
  },
  Object {
    "noise": 0.03247676560000001,
    "order": 104,
    "time": 978338400000,
  },
  Object {
    "noise": 0.02292761656258993,
    "order": 105,
    "time": 978338700000,
  },
  Object {
    "noise": 0.007071379546043151,
    "order": 106,
    "time": 978339000000,
  },
  Object {
    "noise": 0.01575242021726618,
    "order": 107,
    "time": 978339300000,
  },
  Object {
    "noise": 0.025422844900000005,
    "order": 108,
    "time": 978339600000,
  },
  Object {
    "noise": 0.017449817248920863,
    "order": 109,
    "time": 978339900000,
  },
  Object {
    "noise": 0.011416177266187044,
    "order": 110,
    "time": 978340200000,
  },
  Object {
    "noise": 0.014910711212230199,
    "order": 111,
    "time": 978340500000,
  },
  Object {
    "noise": 0.020507862,
    "order": 112,
    "time": 978340800000,
  },
  Object {
    "noise": 0.022699890986330934,
    "order": 113,
    "time": 978341100000,
  },
  Object {
    "noise": 0.026159885050359724,
    "order": 114,
    "time": 978341400000,
  },
  Object {
    "noise": 0.028976724992805764,
    "order": 115,
    "time": 978341700000,
  },
  Object {
    "noise": 0.030213408900000002,
    "order": 116,
    "time": 978342000000,
  },
  Object {
    "noise": 0.022584549471223028,
    "order": 117,
    "time": 978342300000,
  },
  Object {
    "noise": 0.022700950432374106,
    "order": 118,
    "time": 978342600000,
  },
  Object {
    "noise": 0.009169345560431662,
    "order": 119,
    "time": 978342900000,
  },
  Object {
    "noise": -0.0020320948999999963,
    "order": 120,
    "time": 978343200000,
  },
  Object {
    "noise": -0.002994551666187055,
    "order": 121,
    "time": 978343500000,
  },
  Object {
    "noise": 0.0009543377978417297,
    "order": 122,
    "time": 978343800000,
  },
  Object {
    "noise": 0.007870708153237427,
    "order": 123,
    "time": 978344100000,
  },
  Object {
    "noise": 0.01602631490000002,
    "order": 124,
    "time": 978344400000,
  },
  Object {
    "noise": 0.009750405064028786,
    "order": 125,
    "time": 978344700000,
  },
  Object {
    "noise": 0.002389767035971234,
    "order": 126,
    "time": 978345000000,
  },
  Object {
    "noise": -0.010206414982733815,
    "order": 127,
    "time": 978345300000,
  },
  Object {
    "noise": -0.022151524000000002,
    "order": 128,
    "time": 978345600000,
  },
  Object {
    "noise": -0.01700643187769784,
    "order": 129,
    "time": 978345900000,
  },
  Object {
    "noise": -0.009712000287769779,
    "order": 130,
    "time": 978346200000,
  },
  Object {
    "noise": -0.0014165595330935254,
    "order": 131,
    "time": 978346500000,
  },
  Object {
    "noise": 0.012005615399999989,
    "order": 132,
    "time": 978346800000,
  },
  Object {
    "noise": 0.006739798453237411,
    "order": 133,
    "time": 978347100000,
  },
  Object {
    "noise": 0.0010361258733812967,
    "order": 134,
    "time": 978347400000,
  },
  Object {
    "noise": 0.008747103729496393,
    "order": 135,
    "time": 978347700000,
  },
  Object {
    "noise": 0.01470706899999999,
    "order": 136,
    "time": 978348000000,
  },
  Object {
    "noise": 0.014650765230215815,
    "order": 137,
    "time": 978348300000,
  },
  Object {
    "noise": 0.013905155604316544,
    "order": 138,
    "time": 978348600000,
  },
  Object {
    "noise": 0.020368173330935255,
    "order": 139,
    "time": 978348900000,
  },
  Object {
    "noise": 0.026574025200000004,
    "order": 140,
    "time": 978349200000,
  },
  Object {
    "noise": 0.015367897967625901,
    "order": 141,
    "time": 978349500000,
  },
  Object {
    "noise": 0.0069166891366906555,
    "order": 142,
    "time": 978349800000,
  },
  Object {
    "noise": 0.014815386735971228,
    "order": 143,
    "time": 978350100000,
  },
  Object {
    "noise": 0.018151654099999998,
    "order": 144,
    "time": 978350400000,
  },
  Object {
    "noise": 0.021665224453956827,
    "order": 145,
    "time": 978350700000,
  },
  Object {
    "noise": 0.03565166554676259,
    "order": 146,
    "time": 978351000000,
  },
  Object {
    "noise": 0.038246051874100725,
    "order": 147,
    "time": 978351300000,
  },
  Object {
    "noise": 0.042459680000000014,
    "order": 148,
    "time": 978351600000,
  },
  Object {
    "noise": 0.030085305053956832,
    "order": 149,
    "time": 978351900000,
  },
  Object {
    "noise": 0.017542652219424457,
    "order": 150,
    "time": 978352200000,
  },
  Object {
    "noise": 0.017008047426618725,
    "order": 151,
    "time": 978352500000,
  },
  Object {
    "noise": 0.020361840000000006,
    "order": 152,
    "time": 978352800000,
  },
  Object {
    "noise": 0.018198800288489206,
    "order": 153,
    "time": 978353100000,
  },
  Object {
    "noise": 0.016174020189208632,
    "order": 154,
    "time": 978353400000,
  },
  Object {
    "noise": 0.03271316633093527,
    "order": 155,
    "time": 978353700000,
  },
  Object {
    "noise": 0.04980753730000001,
    "order": 156,
    "time": 978354000000,
  },
  Object {
    "noise": 0.028714685057553965,
    "order": 157,
    "time": 978354300000,
  },
  Object {
    "noise": 0.015347221710071954,
    "order": 158,
    "time": 978354600000,
  },
  Object {
    "noise": 0.01181833555971222,
    "order": 159,
    "time": 978354900000,
  },
  Object {
    "noise": 0.011805448900000001,
    "order": 160,
    "time": 978355200000,
  },
  Object {
    "noise": 0.0038640567050359786,
    "order": 161,
    "time": 978355500000,
  },
  Object {
    "noise": -0.0054833127122302094,
    "order": 162,
    "time": 978355800000,
  },
  Object {
    "noise": -0.02803015882158274,
    "order": 163,
    "time": 978356100000,
  },
  Object {
    "noise": -0.0457660256,
    "order": 164,
    "time": 978356400000,
  },
  Object {
    "noise": -0.026954286995683453,
    "order": 165,
    "time": 978356700000,
  },
  Object {
    "noise": -0.012562733399999998,
    "order": 166,
    "time": 978357000000,
  },
  Object {
    "noise": 0.005935393710071929,
    "order": 167,
    "time": 978357300000,
  },
  Object {
    "noise": 0.023053941199999997,
    "order": 168,
    "time": 978357600000,
  },
  Object {
    "noise": 0.026675183286330922,
    "order": 169,
    "time": 978357900000,
  },
  Object {
    "noise": 0.030172225035971237,
    "order": 170,
    "time": 978358200000,
  },
  Object {
    "noise": 0.03882860851870504,
    "order": 171,
    "time": 978358500000,
  },
  Object {
    "noise": 0.0493708642,
    "order": 172,
    "time": 978358800000,
  },
  Object {
    "noise": 0.029309016787769773,
    "order": 173,
    "time": 978359100000,
  },
  Object {
    "noise": 0.00815698541582733,
    "order": 174,
    "time": 978359400000,
  },
  Object {
    "noise": -0.004592897697122303,
    "order": 175,
    "time": 978359700000,
  },
  Object {
    "noise": -0.021154003500000004,
    "order": 176,
    "time": 978360000000,
  },
  Object {
    "noise": -0.023070907512230225,
    "order": 177,
    "time": 978360300000,
  },
  Object {
    "noise": -0.024312762782014392,
    "order": 178,
    "time": 978360600000,
  },
  Object {
    "noise": -0.013531898430215829,
    "order": 179,
    "time": 978360900000,
  },
  Object {
    "noise": -0.008560541000000005,
    "order": 180,
    "time": 978361200000,
  },
  Object {
    "noise": -0.006853857396402874,
    "order": 181,
    "time": 978361500000,
  },
  Object {
    "noise": -0.006560919381294952,
    "order": 182,
    "time": 978361800000,
  },
  Object {
    "noise": -0.018233126945323738,
    "order": 183,
    "time": 978362100000,
  },
  Object {
    "noise": -0.024321641,
    "order": 184,
    "time": 978362400000,
  },
  Object {
    "noise": -0.00219484329856115,
    "order": 185,
    "time": 978362700000,
  },
  Object {
    "noise": 0.015147192402158283,
    "order": 186,
    "time": 978363000000,
  },
  Object {
    "noise": 0.0182183864316547,
    "order": 187,
    "time": 978363300000,
  },
  Object {
    "noise": 0.02946871840000001,
    "order": 188,
    "time": 978363600000,
  },
  Object {
    "noise": 0.019259066419424473,
    "order": 189,
    "time": 978363900000,
  },
  Object {
    "noise": 0.0027269858755395657,
    "order": 190,
    "time": 978364200000,
  },
  Object {
    "noise": 0.00090667142877697,
    "order": 191,
    "time": 978364500000,
  },
  Object {
    "noise": 0.0018819869999999947,
    "order": 192,
    "time": 978364800000,
  },
  Object {
    "noise": -0.0037478268431654707,
    "order": 193,
    "time": 978365100000,
  },
  Object {
    "noise": -0.007336493145323746,
    "order": 194,
    "time": 978365400000,
  },
  Object {
    "noise": -0.009306692766906474,
    "order": 195,
    "time": 978365700000,
  },
  Object {
    "noise": -0.019022767200000006,
    "order": 196,
    "time": 978366000000,
  },
  Object {
    "noise": -0.008572737805755404,
    "order": 197,
    "time": 978366300000,
  },
  Object {
    "noise": 0.003280372474820145,
    "order": 198,
    "time": 978366600000,
  },
  Object {
    "noise": 0.01983713481366907,
    "order": 199,
    "time": 978366900000,
  },
  Object {
    "noise": 0.041537777600000006,
    "order": 200,
    "time": 978367200000,
  },
  Object {
    "noise": 0.01717154048920863,
    "order": 201,
    "time": 978367500000,
  },
  Object {
    "noise": -0.0026842359021582787,
    "order": 202,
    "time": 978367800000,
  },
  Object {
    "noise": -0.01802940003309353,
    "order": 203,
    "time": 978368100000,
  },
  Object {
    "noise": -0.031555235,
    "order": 204,
    "time": 978368400000,
  },
  Object {
    "noise": -0.015323803334532378,
    "order": 205,
    "time": 978368700000,
  },
  Object {
    "noise": -0.0014012288848920806,
    "order": 206,
    "time": 978369000000,
  },
  Object {
    "noise": 0.004635978417266179,
    "order": 207,
    "time": 978369300000,
  },
  Object {
    "noise": 0.007079627899999996,
    "order": 208,
    "time": 978369600000,
  },
  Object {
    "noise": -0.0024305828388489215,
    "order": 209,
    "time": 978369900000,
  },
  Object {
    "noise": -0.01212471828992806,
    "order": 210,
    "time": 978370200000,
  },
  Object {
    "noise": -0.022258401328776982,
    "order": 211,
    "time": 978370500000,
  },
  Object {
    "noise": -0.0374257025,
    "order": 212,
    "time": 978370800000,
  },
  Object {
    "noise": -0.03036144995323741,
    "order": 213,
    "time": 978371100000,
  },
  Object {
    "noise": -0.02854319015107914,
    "order": 214,
    "time": 978371400000,
  },
  Object {
    "noise": -0.03337423042805756,
    "order": 215,
    "time": 978371700000,
  },
  Object {
    "noise": -0.0385382621,
    "order": 216,
    "time": 978372000000,
  },
  Object {
    "noise": -0.030127904743884895,
    "order": 217,
    "time": 978372300000,
  },
  Object {
    "noise": -0.020385755909352512,
    "order": 218,
    "time": 978372600000,
  },
  Object {
    "noise": -0.00729953939064748,
    "order": 219,
    "time": 978372900000,
  },
  Object {
    "noise": 0.008110589000000001,
    "order": 220,
    "time": 978373200000,
  },
  Object {
    "noise": 0.006983145600719429,
    "order": 221,
    "time": 978373500000,
  },
  Object {
    "noise": 0.007260771757553949,
    "order": 222,
    "time": 978373800000,
  },
  Object {
    "noise": -0.005580749989928059,
    "order": 223,
    "time": 978374100000,
  },
  Object {
    "noise": -0.011712596600000007,
    "order": 224,
    "time": 978374400000,
  },
  Object {
    "noise": -0.0032111126884892052,
    "order": 225,
    "time": 978374700000,
  },
  Object {
    "noise": 0.004855598575539577,
    "order": 226,
    "time": 978375000000,
  },
  Object {
    "noise": 0.006268401069784178,
    "order": 227,
    "time": 978375300000,
  },
  Object {
    "noise": 0.003842590300000011,
    "order": 228,
    "time": 978375600000,
  },
  Object {
    "noise": -0.0016397600848920904,
    "order": 229,
    "time": 978375900000,
  },
  Object {
    "noise": -0.007844473435971222,
    "order": 230,
    "time": 978376200000,
  },
  Object {
    "noise": 0.0018221210985611414,
    "order": 231,
    "time": 978376500000,
  },
  Object {
    "noise": 0.012879210000000002,
    "order": 232,
    "time": 978376800000,
  },
  Object {
    "noise": 0.015592804925179851,
    "order": 233,
    "time": 978377100000,
  },
  Object {
    "noise": 0.02352368067194245,
    "order": 234,
    "time": 978377400000,
  },
  Object {
    "noise": 0.020042713756834543,
    "order": 235,
    "time": 978377700000,
  },
  Object {
    "noise": 0.024561694799999992,
    "order": 236,
    "time": 978378000000,
  },
  Object {
    "noise": 0.019771517651798548,
    "order": 237,
    "time": 978378300000,
  },
  Object {
    "noise": 0.011077835964028782,
    "order": 238,
    "time": 978378600000,
  },
  Object {
    "noise": 0.01678370196690647,
    "order": 239,
    "time": 978378900000,
  },
  Object {
    "noise": 0.019042346899999993,
    "order": 240,
    "time": 978379200000,
  },
  Object {
    "noise": 0.017585371496402866,
    "order": 241,
    "time": 978379500000,
  },
  Object {
    "noise": 0.021768620596402874,
    "order": 242,
    "time": 978379800000,
  },
  Object {
    "noise": 0.03609828946115108,
    "order": 243,
    "time": 978380100000,
  },
  Object {
    "noise": 0.04785046900000001,
    "order": 244,
    "time": 978380400000,
  },
  Object {
    "noise": 0.032975570400719426,
    "order": 245,
    "time": 978380700000,
  },
  Object {
    "noise": 0.02274793507194245,
    "order": 246,
    "time": 978381000000,
  },
  Object {
    "noise": 0.01931263595251799,
    "order": 247,
    "time": 978381300000,
  },
  Object {
    "noise": 0.017626006,
    "order": 248,
    "time": 978381600000,
  },
  Object {
    "noise": 0.000979284937410084,
    "order": 249,
    "time": 978381900000,
  },
  Object {
    "noise": -0.019292123280575543,
    "order": 250,
    "time": 978382200000,
  },
  Object {
    "noise": -0.028630788014388488,
    "order": 251,
    "time": 978382500000,
  },
  Object {
    "noise": -0.043085032100000004,
    "order": 252,
    "time": 978382800000,
  },
  Object {
    "noise": -0.030642610122302162,
    "order": 253,
    "time": 978383100000,
  },
  Object {
    "noise": -0.027220376348920862,
    "order": 254,
    "time": 978383400000,
  },
  Object {
    "noise": -0.01859249938848922,
    "order": 255,
    "time": 978383700000,
  },
  Object {
    "noise": -0.013146763500000005,
    "order": 256,
    "time": 978384000000,
  },
  Object {
    "noise": 0.0021993913410071897,
    "order": 257,
    "time": 978384300000,
  },
  Object {
    "noise": 0.0157692010503597,
    "order": 258,
    "time": 978384600000,
  },
  Object {
    "noise": 0.027898421149640293,
    "order": 259,
    "time": 978384900000,
  },
  Object {
    "noise": 0.0490403752,
    "order": 260,
    "time": 978385200000,
  },
  Object {
    "noise": 0.03268201305755396,
    "order": 261,
    "time": 978385500000,
  },
  Object {
    "noise": 0.023655742490647474,
    "order": 262,
    "time": 978385800000,
  },
  Object {
    "noise": 0.02839953893381296,
    "order": 263,
    "time": 978386100000,
  },
  Object {
    "noise": 0.03970360739999999,
    "order": 264,
    "time": 978386400000,
  },
  Object {
    "noise": 0.023272546821582724,
    "order": 265,
    "time": 978386700000,
  },
  Object {
    "noise": 0.015464561743165475,
    "order": 266,
    "time": 978387000000,
  },
  Object {
    "noise": 0.0018263134316546803,
    "order": 267,
    "time": 978387300000,
  },
  Object {
    "noise": -0.012489954599999999,
    "order": 268,
    "time": 978387600000,
  },
  Object {
    "noise": -0.009858466102158271,
    "order": 269,
    "time": 978387900000,
  },
  Object {
    "noise": -0.014160882633093522,
    "order": 270,
    "time": 978388200000,
  },
  Object {
    "noise": -0.014477789352517985,
    "order": 271,
    "time": 978388500000,
  },
  Object {
    "noise": -0.018813266,
    "order": 272,
    "time": 978388800000,
  },
  Object {
    "noise": -0.012617842298561159,
    "order": 273,
    "time": 978389100000,
  },
  Object {
    "noise": -0.007441336749640287,
    "order": 274,
    "time": 978389400000,
  },
  Object {
    "noise": 0.002157432741007187,
    "order": 275,
    "time": 978389700000,
  },
  Object {
    "noise": 0.007061178700000004,
    "order": 276,
    "time": 978390000000,
  },
  Object {
    "noise": 0.0056866901884892185,
    "order": 277,
    "time": 978390300000,
  },
  Object {
    "noise": 0.002258690253237411,
    "order": 278,
    "time": 978390600000,
  },
  Object {
    "noise": -0.019674819176258993,
    "order": 279,
    "time": 978390900000,
  },
  Object {
    "noise": -0.0369286096,
    "order": 280,
    "time": 978391200000,
  },
  Object {
    "noise": -0.02801903755611511,
    "order": 281,
    "time": 978391500000,
  },
  Object {
    "noise": -0.023191657288489213,
    "order": 282,
    "time": 978391800000,
  },
  Object {
    "noise": -0.03285706549280576,
    "order": 283,
    "time": 978392100000,
  },
  Object {
    "noise": -0.042851204000000004,
    "order": 284,
    "time": 978392400000,
  },
  Object {
    "noise": -0.036292517128057554,
    "order": 285,
    "time": 978392700000,
  },
  Object {
    "noise": -0.03082181899280576,
    "order": 286,
    "time": 978393000000,
  },
  Object {
    "noise": -0.01979200023165468,
    "order": 287,
    "time": 978393300000,
  },
]
`;
